### PR TITLE
Tooltips, more styling, bug fixes, documentation

### PR DESCRIPTION
**Pull request by KrisSiegel**
This PR accomplishes the following

- Added tooltips. These are called html sprites in that they're html (that can be styled) but display based on canvas calculations. @michaelkunyunghoward styled the tooltips here
- Changed example pages around so there is one for loading each file required (for devs), one for loading concat sources and one for minified sources
- Added a static server for developers wanting to jump in and change things
- Added a config and related event so config changes can be subscribed to and used (tooltips use this to enable and disable themselves). Currently only tooltips are exposed here
- Revised documentation to outline options and running examples

After this I'm wrapping up the BoData side then later on I'll come back to do a little bit of tidying up (expose all properties via config and make pattern uniform squares as per Mike).

@immuta/developers

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,8 +8,19 @@ You can grab BoPattern.js in one of two ways.
...
 ## Quick start
-Simply set this folder to be served through a static web server then navigate to ```examples/```. This will load up the example page showing BoPattern.js working with a button that generates a random data set. Simple!
+Simply set this folder to be served through a static web server then navigate to any of the HTML files contained within ```examples/```. This will load up the example page showing BoPattern.js working with a button that generates a random data set. Simple!
+
+## Running the examples
+There is an included static file server when you pull down the repo that can be started by running ```npm start``` (**note** you must first run ```npm install```). Each of the example files are available at:
+
+```http://localhost:8080/examples/example-all-sources.html``` -> This page manually includes each JavaScript and CSS file required to run BoPattern.js. Use this if you're a developer and you want to modify BoPattern.js without rebuilding it constantly.
+
+```http://localhost:8080/examples/example-concat-sources.html``` -> This page includes only the concatenated sources. This is handle should you want to debug something but not develop against it.
```
Comment (barryhammen) [inline README.md]: 'This is handle'...

Comment (KrisSiegel) [inline README.md]: English hard somtims

**Comment by barryhammen** [comment]
:+1:

File: src/utilities/loader.js [hunk 2]
```diff
@@ -31,7 +31,7 @@ BoPattern.extend(function(internal) {
                             var datum = data[i][j];
                             if (msngr.isObject(datum)) {
                                 datum.value = datum.value || datum.data;
-                                datum.label = datum.label || datum.value;
+                                datum.label = datum.label || datum.value.toFixed(2);
```
Comment (KyleLilly) [inline src/utilities/loader.js]: I know we moved a bunch of stuff to configuration based values maybe at some point the number of digits for the label should be configurable.

File: src/loops/update-loop.js [hunk 3]
```diff
@@ -47,6 +47,10 @@ BoPattern.extend(function(internal) {
             // This sets the location of the bounded area
             internal.boundedX1 = ((internal.screenWidth - internal.boundedWidth) / 2);
             internal.boundedY1 = (((internal.screenHeight - internal.boundedHeight) / 2) + (((internal.screenHeight - internal.boundedHeight) / 2) / 2));
+
+            // This sets the absolute location of elements as per the whole window (not just the cancas)
```
Comment (KyleLilly) [inline src/loops/update-loop.js]: Canvas not `cancas`.

File: dist/bopattern.js [hunk 4]
```diff
@@ -694,7 +811,7 @@ BoPattern.extend(function(internal) {
                             var datum = data[i][j];
                             if (msngr.isObject(datum)) {
                                 datum.value = datum.value || datum.data;
-                                datum.label = datum.label || datum.value;
+                                datum.label = datum.label || datum.value.toFixed(2);
```
Comment (KyleLilly) [inline dist/bopattern.js]: Same as other spot - in the future would be nice if this was a config option.

**Comment by KyleLilly** [comment]
Minor issues so :+1:

File: src/loops/update-loop.js (hunk 3 above)
Comment (KrisSiegel) [inline src/loops/update-loop.js]: image

File: src/utilities/loader.js (hunk 2 above)
Comment (KrisSiegel) [inline src/utilities/loader.js]: This is only used in the event that someone doesn't supply a label. They should really supply a label. So I honestly don't care too much here :)

**Pull request merged**